### PR TITLE
Add documentation on how to deal with method signatures.

### PR DESCRIPTION
**Pull request by sjoerdjob**
This took me some time to figure out, so I thought it would be good to have for other developers.

### CODE REVIEW

File: docs/internals/index.rst [hunk 1]
```diff
@@ -8,3 +8,4 @@ Project Internals
    contributing
    releases
    roadmap
+   signatures
```
Comment (sjoerdjob) [inline docs/internals/index.rst]: I can easily imagine you'd prefer having this somewhere else. Does anybody have a good suggestion?

File: docs/internals/signatures.rst [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                val = other;
+            }
+        }
+        return val;
+    }
+
+A function accepting keyword arguments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Of course, sometimes you want to use keyword arguments. Unfortunately, that is
```
Comment (sjoerdjob) [inline docs/internals/signatures.rst]: The `kwargs` is not really used in the codebase, so I am a bit edgy before documenting how it should be used when there is no example to look at.

File: docs/internals/index.rst (hunk 1 above)
Comment (freakboy3742) [inline docs/internals/index.rst]: It's as good as anywhere for now.

File: docs/internals/signatures.rst (hunk 2 above)
Comment (freakboy3742) [inline docs/internals/signatures.rst]: It's an odd edge case; I'm happy to commit this as is, and I'll flesh out the rest at some point.

**Pull request merged**